### PR TITLE
Bluetooth: Controller: Remove legacy BT_CTLR_FAST_ENC option

### LINKED ISSUE #63212: Build warning on entropy_nrf5 with clang
**Describe the bug**

drivers/entropy/entropy_nrf5.c is pushing gcc build options with pragmas without guarding for gcc.
When building with clang this will cause a warning (and the option be ignored)

**To Reproduce**
1. mkdir build; cd build
2. ZEPHYR_TOOLCHAIN_VARIANT=llvm cmake -GNinja -DBOARD=nrf52_bsim ../tests/bsim/bluetooth/ll/conn/ -DCONF_FILE=prj_split.conf
3. ninja

**Expected behavior**
No warning when building with clang

**Impact**
When building with clang, warning on build which can result on build error if warning are treated as errors

**Logs and console output**
```
drivers/entropy/entropy_nrf5.c:119:13: warning: unknown pragma ignored [-Wunknown-pragmas]
#pragma GCC push_options
            ^
drivers/entropy/entropy_nrf5.c:178:13: warning: unknown pragma ignored [-Wunknown-pragmas]
#pragma GCC pop_options
            ^
```

**Environment (please complete the following information):**
 - OS: Linux, Unbuntu 22.04
 - Toolchain (e.g Zephyr SDK, ...): LLVM 14
 - Commit SHA or Version used: Today's main 1b0cb324f3827ed5dcd08b63de7b8168fc2a3ffa

**Additional info**
Since fb3387a4907

### PR DESCRIPTION
**Pull request by carlescufi**
When the legacy LLCP implementation was removed this Kconfig option was
mistakenly left over. Remove it now with all its users.  

Fixes #63212.

### CODE REVIEW

File: drivers/entropy/entropy_nrf5.c [hunk 1]
```diff
@@ -116,10 +116,12 @@ static int random_byte_get(void)
 	return retval;
 }
 
+#if defined(__GNUC__)
 #pragma GCC push_options
 #if defined(CONFIG_BT_CTLR_FAST_ENC)
```
Comment (gmarull) [inline drivers/entropy/entropy_nrf5.c:121]: `CONFIG_BT_CTLR_FAST_ENC` should likely be documented to be GCC-only (I think we can't create toolchain dependencies in Kconfig)

Comment (gmarull) [reply drivers/entropy/entropy_nrf5.c:121]: btw, shouldn't we have:

```
#if defined(CONFIG_BT_CTLR_FAST_ENC) && defined(__GNUC__)
#pragma GCC push_options
#pragma GCC optimize ("Ofast")
#endif /* defined(CONFIG_BT_CTLR_FAST_ENC) && defined(__GNUC__) */
```

**Review by aescolar — DISMISSED** [review]
I'm fine with this or gmarull's proposal

File: drivers/entropy/entropy_nrf5.c (hunk 1 above)
Comment (cvinayak) [inline drivers/entropy/entropy_nrf5.c:121]: CONFIG_BT_CTLR_FAST_ENC  is stale Kconfig.... we should just remove this use of pragma and the Kconfig.

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: drivers/entropy/entropy_nrf5.c (hunk 1 above)
Comment (carlescufi) [reply drivers/entropy/entropy_nrf5.c]: oh I see. OK then, let's get rid of it.

**Review by jhedberg — APPROVED** [review]

**Review by erwango — APPROVED** [review]
On STM32 driver this was never used actually.
A dumb copy/paste I guess. Thanks for the clean up!

**Review by cvinayak — APPROVED** [review]

**Review by aescolar — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Pull request merged**